zerver/lib/actions.py

```diff
@@ -1033,6 +1043,18 @@ def should_send(setting: str, row: Dict[str, Any]) -> bool:
             if should_send('email_notifications', row)
         } - user_ids_muting_topic
 
+        wildcard_mention_email_user_ids = {
+            row['user_profile_id']
+            for row in subscription_rows
+            if not row['is_muted'] and row['user_profile_wildcard_mention_email_notifications']
```

Comment:
With the new design, these checks should become:
```
if not row['is_muted'] and row['user_profile_wildcard_mentions_notify']
```

Though again we're probably inverting the meaning and/or wanting to only construct these lists in the event